From 8ea11d645dda5895bbe5162803b5f6ef8573d6ea Mon Sep 17 00:00:00 2001
From: Mohammad Athari Bin Ismail <mohammad.athari.ismail@intel.com>
Date: Thu, 13 Aug 2020 17:04:14 +0800
Subject: [PATCH] REVERTME: net: stmmac: Disable Split Header for TGL and EHL

There is an issue found where MAC is not able to split header for
IP non-TCP/UDP with SPLM of L3L4(0x0) and COMBN(0x2). So, disable the
Split Header for TGL and EHL until the issue is fixed.

Suggested-by: Ong, Boon Leong <boon.leong.ong@intel.com>
Signed-off-by: Mohammad Athari Bin Ismail <mohammad.athari.ismail@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 2 +-
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c  | 1 +
 include/linux/stmmac.h                            | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 6379ab33ca62..8e57fad3f3fd 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -6774,7 +6774,7 @@ int stmmac_dvr_probe(struct device *device,
 		dev_info(priv->device, "TSO feature enabled\n");
 	}
 
-	if (priv->dma_cap.sphen) {
+	if (priv->dma_cap.sphen && priv->plat->sph_en) {
 		ndev->hw_features |= NETIF_F_GRO;
 		priv->sph = true;
 		dev_info(priv->device, "SPH feature enabled\n");
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index b5633f71a870..7b74a4fb4a77 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -187,6 +187,7 @@ static int intel_mgbe_common_data(struct pci_dev *pdev,
 	plat->tsn_est_en = 1;
 	plat->tsn_fpe_en = 1;
 	plat->tsn_tbs_en = 1;
+	plat->sph_en = 0;
 	/* FPE HW Tunable */
 	plat->fprq = 1;
 	plat->afsz = 0;  /* Adjustable Fragment Size */
diff --git a/include/linux/stmmac.h b/include/linux/stmmac.h
index c9e4cbe691d3..fde9726318ca 100644
--- a/include/linux/stmmac.h
+++ b/include/linux/stmmac.h
@@ -195,6 +195,7 @@ struct plat_stmmacenet_data {
 	bool tsn_est_en;
 	bool tsn_fpe_en;
 	bool tsn_tbs_en;
+	bool sph_en;
 	int mac_port_sel_speed;
 	bool en_tx_lpi_clockgating;
 	int has_xgmac;
-- 
2.17.1

